dm/worker/config.go

```diff
@@ -266,26 +315,40 @@ func (c *Config) adjustFlavor() error {
 			return terror.ErrNotSupportedFlavor.Generate(c.Flavor)
 		}
 	}
-	// decrypt password
-	clone, err := c.DecryptPassword()
-	if err != nil {
-		return err
+
+	c.Flavor, err = utils.GetFlavor(ctx, db)
+	if ctx.Err() != nil {
+		err = terror.Annotatef(err, "time cost to get flavor info exceeds %s", dbGetTimeout)
+	}
+	return terror.WithScope(err, terror.ScopeUpstream)
+}
+
+func (c *Config) adjustServerID(ctx context.Context, db *sql.DB) error {
+	if c.ServerID != 0 {
+		return nil
+	}
+
+	serverIDs, err := getAllServerIDFunc(ctx, db)
+	if ctx.Err() != nil {
+		err = terror.Annotatef(err, "time cost to get server-id info exceeds %s", dbGetTimeout)
 	}
-	from := clone.From
-	from.RawDBCfg = config.DefaultRawDBConfig().SetReadTimeout(flavorReadTimeout)
-	fromDB, err := applyNewBaseDB(from)
 	if err != nil {
 		return terror.WithScope(err, terror.ScopeUpstream)
 	}
-	defer fromDB.Close()
 
-	ctx, cancel := context.WithTimeout(context.Background(), flavorGetTimeout)
-	defer cancel()
-	c.Flavor, err = utils.GetFlavor(ctx, fromDB.DB)
-	if ctx.Err() != nil {
-		err = terror.Annotatef(err, "time cost to get flavor info exceeds %s", flavorGetTimeout)
+	rand.Seed(time.Now().UnixNano())
```

Comment:
can we do this when starting the process?

Reply:
address comment